terraform/protoc-gen-terraform-teleport.yaml

```diff
@@ -135,25 +135,29 @@ exclude_fields:
     - "UserSpecV2.Status"
 
 name_overrides:
-    # Teleport API has a new field: RedirectURLs which maps to `redirect_url` - an already existing field name
-    # This `name_overrides` entry will override the name for RedirectURLs: `redirect_urls`
+    # Teleport API has two new fields: AuthPreference.Spec.RequireMFAType and Role.Spec.Options.RequireMFAType
+    # The following explanation is valid for both fields, but we only mention the first one so that's easier to understand.
+    #
+    # Teleport API has a new field: AuthPreference.Spec.RequireMFAType which maps to `require_session_mfa` - an already existing field name
+    # This `name_overrides` entry will override the name for RequireMFAType: `require_mfa_type`
     # This way, users can use both:
-    # - `redirect_url` for a single string: now deprecated (https://github.com/gravitational/teleport/pull/13046)
-    # - `redirect_urls` for a list of strings: the recommended way from now on
+    # - `require_session_mfa` for a boolean: now deprecated (https://github.com/gravitational/teleport/pull/16034)
+    # - `require_mfa_type` an enum OFF|SESSION|SESSION_AND_HARDWARE_KEY|HARDWARE_KEY_TOUCH
+    #     the recommended setting https://github.com/gravitational/teleport/blob/1edafaab3360f42f037a81c7baeabfa0627a7b66/api/proto/teleport/legacy/types/types.proto#L4325-L4338
 
     # Going forward we want to:
-    # - change the type of `redirect_url` from string into list of strings to match the resource's yaml representation in Teleport
-    # - after doing the above, remove `redirect_urls` field
+    # - change the type of `require_session_mfa` from boolean into an enum to match the resource's yaml representation in Teleport
+    # - after doing the above, remove `require_mfa_type` field
     
-    # When we change the type of `redirect_url` from string to list of strings we break user's configurations
+    # When we change the type of `require_session_mfa` from boolean to enum we break user's configurations
     # They'll need to change
-    # redirect_url: "https://example.org/redirect"
```

Comment:
@Joerger Did we remove support for singular `redirect_url` field in Teleport 11?

Reply:
Teleport 11 Still supports `redirect_url` for JSON marshalling, aka for dynamic resources and file conf. Only Terraform is impacted by this change since it does not use JSON marshaling and therefore does not respect the `github.com/gravitational/teleport/api/types/wrappers.Strings` type.

Here's the recent commit that deprecates the `RedirectURL` proto field - [cc5931d1014915f641b232956901df6b9cbbe08a](https://github.com/gravitational/teleport/pull/17047/commits/2a12068cd01b79f149b1731fcf75d6827ee41a3f)